console/console-init/ui/src/modules/device/utils/utils.ts

```diff
@@ -0,0 +1,29 @@
+import { IDeviceFilterCriteria } from "../components";
+
+const getFilterCriteriaIndex = (
+  criteriaList: IDeviceFilterCriteria[],
+  criteria: IDeviceFilterCriteria
+) => {
+  for (let i = 0; i < criteriaList.length; i++) {
```

Comment:
not sure purpose of this function and looping over there but i would like to suggest one point here.
Don't calculate length of the array in loop , it cause of performance, it will calculate length every time. best way is calculate length before loop and use that variable.
const criteriaLength=criteriaList.length;
for (let i = 0; i < criteriaLength; i++) {
}